app/src/main/java/com/fictadvisor/android/ui/LoginFragment.kt

```diff
@@ -0,0 +1,125 @@
+package com.fictadvisor.android.ui
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.Navigation
+import com.fictadvisor.android.R
+import com.fictadvisor.android.data.dto.AuthLoginResponse
+import com.fictadvisor.android.data.dto.BaseResponse
+import com.fictadvisor.android.databinding.FragmentLoginBinding
+import com.fictadvisor.android.repository.AuthRepository
+import com.fictadvisor.android.utils.StorageUtil
+import com.fictadvisor.android.validator.LoginInputValidator
+import com.fictadvisor.android.viewmodel.AuthViewModel
+import com.fictadvisor.android.viewmodel.AuthViewModelFactory
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class LoginFragment : Fragment() {
+    private lateinit var binding: FragmentLoginBinding
+    private lateinit var authViewModel: AuthViewModel
+    private val authRepository = AuthRepository()
+    private lateinit var inputValidator: LoginInputValidator
+    private lateinit var storageUtil: StorageUtil
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentLoginBinding.inflate(inflater, container, false)
+        val view = binding.root
+        inputValidator = LoginInputValidator(requireContext())
+        storageUtil = StorageUtil(requireContext())
+
+        authViewModel = ViewModelProvider(
+            this,
+            AuthViewModelFactory(authRepository)
+        ).get(AuthViewModel::class.java)
+
+        binding.buttonLogin.setOnClickListener {
+            val username = binding.editTextLogin.text.toString()
+            val password = binding.editTextPassword.text.toString()
+            if (inputValidator.isLoginDataValid(username)) {
+                loginUser(username, password)
+            }
+        }
+
+        binding.buttonPrevious.setOnClickListener {
```

Comment:
Let's just stick to the same style and make ClickListeners as seperate functions